examples/trusted_database/module_1/rust/src/lib.rs

```diff
@@ -0,0 +1,142 @@
+            let err = grpc::build_status(grpc::Code::InvalidArgument, &NO_LOCATION_ERROR);
+            error!("{:?}", err);
+            err
+        })?;
+        let nearest_point = self.points_of_interest.iter().fold(
+            (None, f32::MAX),
+            |(closest_point, min_distance), point| {
+                let point_location = point.location.clone().expect("Non-existing location");
+                let distance = distance(location.clone(), point_location);
+                if distance < min_distance {
+                    (Some(point.clone()), distance)
+                } else {
+                    (closest_point, min_distance)
+                }
+            },
+        );
+        match nearest_point.0 {
+            Some(point) => {
+                debug!("Found the nearest Point Of Interest: {:?}", point);
+                Ok(ListPointsOfInterestResponse {
+                    point_of_interest: Some(point),
+                })
+            }
+            None => {
+                let err = grpc::build_status(grpc::Code::Internal, &EMPTY_DATABASE_ERROR);
+                error!("{:?}", err);
+                Err(err)
+            }
+        }
+    }
+}
+
+/// Returns a distance (in kilometers) between two locations using the Haversine formula
+/// (ignoring height variations):
+/// https://en.wikipedia.org/wiki/Haversine_formula
+pub fn distance(first: Location, second: Location) -> f32 {
+    let first_latitude_radians = first.latitude.to_radians();
+    let second_latitude_radians = second.latitude.to_radians();
+
+    let latitude_difference_radians = (first.latitude - second.latitude).to_radians();
+    let longitude_difference_radians = (first.longitude - second.longitude).to_radians();
+
+    let central_angle = 2.0
+        * ((latitude_difference_radians / 2.0).sin().powi(2)
+            + (first_latitude_radians.cos()
+                * second_latitude_radians.cos()
+                * (longitude_difference_radians / 2.0).sin().powi(2)))
+        .sqrt()
+        .asin();
+
+    // Earth radius in kilometers.
+    let earth_radius = 6371.0_f32;
+
+    // Compute distance.
+    earth_radius * central_angle
+}
+
+oak::entrypoint!(handler_oak_main => |in_channel| {
+    oak::logger::init_default();
+
+    // Receive command.
+    let command_receiver = oak::io::Receiver::<TrustedDatabaseCommand>::new(in_channel);
+    let command: TrustedDatabaseCommand =
+        command_receiver.receive().expect("Couldn't receive command");
+    let receiver = command.invocation_receiver.expect("Couldn't receive gRPC invocation receiver");
+
+    // Run event loop and handle incoming invocations.
+    let node = TrustedDatabaseHandlerNode { points_of_interest: command.points_of_interest };
+    let dispatcher = TrustedDatabaseDispatcher::new(node);
+    let invocation_receiver = receiver.receiver.expect("Empty gRPC invocation receiver");
+    oak::run_event_loop(dispatcher, invocation_receiver);
```

Comment:
Should this run the event loop? If a new node is generated for every invocation, it should probably just read the one invocation, write the response and then terminate. Or ar least make the comments very clear that the event loop will only run once because the invocation channel is closed by the node that created it. Otherwise it might be confusing to people who read through the example.

Reply:
I used `run_event_loop` in order to use automatically generated `TrustedDatabaseDispatcher` and not implement `handler` by hand.
Added a comment.